checkov/terraform/graph_builder/variable_rendering/renderer.py

```diff
@@ -323,28 +323,41 @@ def _process_dynamic_blocks(dynamic_blocks: list[dict[str, Any]] | dict[str, Any
             if not block_content or not dynamic_values:
                 continue
 
-            dynamic_value_ref = f"{block_name}.value"
+            dynamic_value_dot_ref = f"{block_name}.value"
+            dynamic_value_bracket_ref = f'{block_name}["value"]'
             dynamic_arguments = [
                 argument
                 for argument, value in block_content.items()
-                if value == dynamic_value_ref or isinstance(value, str) and dynamic_value_ref in value
+                if (value == dynamic_value_dot_ref or value == dynamic_value_bracket_ref) or isinstance(value, str) and dynamic_value_dot_ref in value
             ]
 
             if dynamic_arguments:
                 block_confs = []
                 for dynamic_value in dynamic_values:
                     block_conf = deepcopy(block_content)
+                    block_conf.pop('dynamic', None)
                     for dynamic_argument in dynamic_arguments:
                         if dynamic_type == DYNAMIC_BLOCKS_MAPS:
                             if not isinstance(dynamic_value, dict):
                                 continue
-                            dynamic_value_in_map = TerraformVariableRenderer.extract_dynamic_value_in_map(block_content[dynamic_argument])
-                            block_conf[dynamic_argument] = dynamic_value[dynamic_value_in_map]
+                            dynamic_value_in_map = TerraformVariableRenderer.extract_dynamic_value_in_map(
+                                block_content[dynamic_argument]
+                            )
+                            if block_name not in dynamic_value:
+                                block_conf[dynamic_argument] = dynamic_value[dynamic_value_in_map]
+                            else:
+                                block_conf[dynamic_argument] = dynamic_value[block_name][0][dynamic_value_in_map]
                         else:
                             block_conf[dynamic_argument] = dynamic_value
 
                     block_confs.append(block_conf)
-                rendered_blocks[block_name] = block_confs
+                rendered_blocks[block_name] = block_confs if len(block_confs) > 1 else block_confs[0]
+
+            if 'dynamic' in block_content:
+                next_key = list(block_content['dynamic'].keys())[0]
```

Comment:
```suggestion
                next_key = next(iter(block_content['dynamic'].keys()))
```
no need to create a list, this does the same

Reply:
@ChanochShayner `next` throws `StopIteration`, need to catch it if you choose to accept @gruebel's suggestion

Reply:
that's true, but the current implementation would also throw an error (IndexError), if the `dynamic` dict is empty. but good idea to do some `try ... except` around it.